packages/installer/cos.sh

```diff
@@ -539,47 +691,47 @@ validate_device()
 ## UPGRADER
 
 find_partitions() {
-    STATE=$(blkid -L COS_STATE || true)
-    if [ -z "$STATE" ]; then
+    _STATE=$(blkid -L COS_STATE || true)
+    if [ -z "$_STATE" ]; then
         echo "State partition cannot be found"
         exit 1
     fi
 
-    OEM=$(blkid -L COS_OEM || true)
+    _OEM=$(blkid -L COS_OEM || true)
 
-    PERSISTENT=$(blkid -L COS_PERSISTENT || true)
-    if [ -z "$PERSISTENT" ]; then
+    _PERSISTENT=$(blkid -L COS_PERSISTENT || true)
+    if [ -z "$_PERSISTENT" ]; then
         echo "Persistent partition cannot be found"
         exit 1
     fi
 
-    COS_ACTIVE=$(blkid -L COS_ACTIVE || true)
-    if [ -n "$COS_ACTIVE" ]; then
-        CURRENT=active.img
+    _COS_ACTIVE=$(blkid -L COS_ACTIVE || true)
+    if [ -n "$_COS_ACTIVE" ]; then
+        _CURRENT=active.img
     fi
 
-    COS_PASSIVE=$(blkid -L COS_PASSIVE || true)
-    if [ -n "$COS_PASSIVE" ]; then
-        CURRENT=passive.img
+    _COS_PASSIVE=$(blkid -L COS_PASSIVE || true)
+    if [ -n "$_COS_PASSIVE" ]; then
+        _CURRENT=passive.img
     fi
 
-    if [ -z "$CURRENT" ]; then
+    if [ -z "$_CURRENT" ]; then
         # We booted from an ISO or some else medium. We assume we want to fixup the current label
         read -p "Could not determine current partition. Do you want to overwrite your current active partition? (CURRENT=active.img) [y/N] : " -n 1 -r
```

Comment:
Do we need to update the help message ?

Reply:
why? am Im missing something here? Is CURRENT able to be overwritten? 'cause its not documented

Reply:
`    # We booted from an ISO or some else medium. We assume we want to fixup the current label
        read -p "Could not determine current partition. Do you want to overwrite your current active partition? (CURRENT=active.img) [y/N] : 
`

it is few lines below if no `CURRENT` was found. Basically that is the only way to recover other images (besides active) for e.g. when booting from a LiveCD

Reply:
Sorry, Im still not getting the problem here, could you elaborate further?

Is current supposed to be able to overwrite via env var?

Reply:
yep, the script tries to find a "current" image as target for the upgrade process but it is a best effort, hence why the command prompt. Since the prompt could break automatic flows (scripts, etc) it is meant to be possible to set it to avoid any interaction

Reply:
ah, ok. Do we support the CURRENT in the config file or only via env vars?

Reply:
I don't think it makes sense really sense in the config file as it is a corner-case here. Besides defaults wouldn't be relevant as we try to auto-detect it

Reply:
updated!